src/main/java/io/opentracing/contrib/global/GlobalTracer.java

```diff
@@ -0,0 +1,181 @@
+     * <li>A Tracer implementation can be automatically provided using the Java {@link ServiceLoader} through the
+     * <code>META-INF/services/io.opentracing.Tracer</code> service definition file.<br>
+     * The {@linkplain GlobalTracer} class will not attempt to choose between implementations;
+     * if more than one is found by the {@link ServiceLoader service loader},
+     * a warning is logged and tracing is disabled by falling back to the default implementation:</li>
+     * <li>If no tracer implementation is available, the {@link io.opentracing.NoopTracer NoopTracer}
+     * will be used.</li>
+     * </ol>
+     *
+     * @return The global tracer.
+     */
+    public static Tracer get() {
+        return INSTANCE;
+    }
+
+    /**
+     * Explicit registration of a configured {@link Tracer} to back the behaviour
+     * of the {@link #get() global tracer}.
+     * <p>
+     * The previous global tracer is returned so it can be restored later if necessary.
+     *
+     * @param tracer Tracer to use as global tracer.
+     * @return The previous global tracer.
+     */
+    public static Tracer set(final Tracer tracer) {
+        if (tracer instanceof GlobalTracer) {
+            throw new IllegalArgumentException("Attempted to set the GlobalTracer as delegate of itself.");
+        }
+        Tracer previous = INSTANCE.globalTracer.getAndSet(tracer);
+        logChangedTracer(tracer, previous);
+        return previous;
+    }
+
+    /**
+     * Wraps the {@link Callable} to execute within a new {@link Span} if an
+     * {@link TracedCallable#withOperationName(String) operationName} is also specified.<br>
+     * If no operationName is provided, the callable will execute as-is without starting a new span.
+     *
+     * @param operationName Name of the traced operation.
+     * @param callable      Callable to wrap.
+     * @param <V>           Return type of the wrapped call.
+     * @return The wrapped call.
+     * @see TracedCallable#withOperationName(String)
+     */
+    public static <V> TracedCallable<V> traced(String operationName, Callable<V> callable) {
+        return TracedCallable.of(callable).withOperationName(operationName);
+    }
+
+    /**
+     * Wraps the {@link Runnable} to execute within a new {@link Span} if an
+     * {@link TracedCallable#withOperationName(String) operationName} is also specified.<br>
+     * If no operationName is provided, the callable will execute as-is without starting a new span.
+     *
+     * @param operationName Name of the traced operation.
+     * @param runnable      Runnable to wrap.
+     * @return The wrapped call.
+     * @see TracedRunnable#withOperationName(String)
+     */
+    public static TracedRunnable traced(String operationName, Runnable runnable) {
+        return TracedRunnable.of(runnable);
+    }
+
+    @Override
+    public SpanBuilder buildSpan(String operationName) {
```

Comment:
While I don't recommend using GlobalTracer.get() as a constant anywhere, this is a library and I cannot prohibit using it this way. Especially since tracing is similar to logging in my mind and that's how people are already used to working with loggers.
This way, the behaviour of the global tracer is known before and after application initialization for all references (obtained either before or after initialization completed). Note due to the atomic reference, for observer of GlobalTracer there is no 'during initialization' which would make things really unpredictable.
A second advantage is that it forces people to stick to the API and not take implementation-specific shortcuts, because that's all they get from the singleton tracer.
Thirdly, it saves me from explaining why the GlobalTracer is not a Tracer, as you correctly mentioned previously 😉 .